src/codegen/compvars.cpp

```diff
@@ -74,43 +74,62 @@ std::string ValuedCompilerType<llvm::Value*>::debugName() {
     return rtn;
 }
 
+// Returns a CompilerVariable representing the given variable's type
+static CompilerVariable* getClassCompilerVariable(IREmitter& emitter, ConcreteCompilerVariable* var) {
+    // val->cls
+    static_assert(offsetof(Box, cls) == 0, "");
+    llvm::Value* cls_ptr = emitter.getBuilder()->CreateConstInBoundsGEP2_32(var->getValue(), 0, 0);
+    llvm::Value* cls = emitter.getBuilder()->CreateLoad(cls_ptr);
+    assert(cls->getType() == g.llvm_class_type_ptr);
+    // cast from BoxedClass* to Box*
+    llvm::Value* cls_as_box = emitter.getBuilder()->CreateGEP(
+        cls, { llvm::ConstantInt::get(g.i32, 0), llvm::ConstantInt::get(g.i32, 0), llvm::ConstantInt::get(g.i32, 0) });
+    assert(cls_as_box->getType() == g.llvm_value_type_ptr);
+    return new ConcreteCompilerVariable(UNKNOWN, cls_as_box, false);
+}
```

Comment:
I think this is probably better as a method of the CompilerType class, since for a bunch of compiler types we will know exactly what the class is and they can override the behavior.  can maybe punt on that for now since we might not need/want this function.